src/sdk/nsfs_glacier_backend/backend.js

```diff
@@ -151,8 +133,7 @@ class GlacierBackend {
         // in races
         if (
             stat.xattr[GlacierBackend.XATTR_RESTORE_REQUEST] ||
-            stat.xattr[GlacierBackend.XATTR_RESTORE_EXPIRY] ||
-            stat.xattr[GlacierBackend.XATTR_RESTORE_REQUEST_STAGED]) {
+            stat.xattr[GlacierBackend.XATTR_RESTORE_EXPIRY]) {
```

Comment:
I think what we want is to move the _get_restore_status function from namspace_fs to here, and reuse it also for resolving these states of should_restore and should_migrate - WDYT?

Reply:
Another small note - a few lines above this one there is a hardcoded 'GLACIER' string being used - I think we should avoid repeating these literals to avoid typos and use a constant. Perhaps look for this literal in the rest of the code and see if we can replace all those uses.

Reply:
get_restore_status() should check the storage_class too
and should_migrate can call it to check if state is CAN_RESTORE
and should_restore can call it to check if state is ONGOING